packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -0,0 +1,353 @@
+        this.table.isColumnBeingSized = true;
+        document.addEventListener('mousemove', this.onDividerMouseMove);
+        document.addEventListener('mouseup', this.onDividerMouseUp);
+    }
+
+    public updateTableViewportMinWidth(): void {
+        if (!this.table.$fastController.isConnected) {
+            return;
+        }
+
+        this.table.tableViewportMinWidth = Math.round(
+            (this.table.headerRowActionContainer?.getBoundingClientRect()
+                .width ?? 0)
+                + this.getAllColumnsMinimumWidth()
+                + this.table.virtualizer.headerContainerMarginRight
+        );
+    }
+
+    public handleChange(source: unknown, args: unknown): void {
+        if (
+            source instanceof Virtualizer
+            && args === 'headerContainerMarginRight'
+        ) {
+            this.updateTableViewportMinWidth();
+        }
+    }
+
+    private readonly onDividerMouseMove = (event: Event): void => {
+        const mouseEvent = event as MouseEvent;
+        const deltaX = mouseEvent.movementX > 0
+            ? Math.floor(mouseEvent.movementX)
+            : Math.ceil(mouseEvent.movementX);
+        this.currentTotalDelta += deltaX;
+        for (let i = 0; i < this.table.columns.length; i++) {
+            this.table.columns[i]!.columnInternals.currentPixelWidth = this.initialColumnPixelWidths[i]?.initialPixelWidth;
+        }
+        this.currentTotalDelta = this.pinColumnSizeDelta(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+        this.performCascadeSizeLeft(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+        this.performCascadeSizeRight(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+
+        this.table.tableWidthFactor = Math.max(
+            this.getCurrentTotalTableWidth() / this.initialTableWidth!,
+            1
+        );
+    };
+
+    private readonly onDividerMouseUp = (): void => {
+        document.removeEventListener('mousemove', this.onDividerMouseMove);
+        document.removeEventListener('mouseup', this.onDividerMouseUp);
+        this.unflagActiveColumnDividers();
+        this.updateTableViewportMinWidth();
+        this.resetGridSizedColumns();
+        this.table.isColumnBeingSized = false;
+    };
+
+    private getAllColumnsMinimumWidth(): number {
+        let totalColumnMiniumWidth = 0;
+        for (const column of this.table.columns) {
+            totalColumnMiniumWidth += column.columnInternals.minPixelWidth;
+        }
+        return totalColumnMiniumWidth;
+    }
+
+    private getTotalColumnFixedWidth(): number {
+        let totalColumnFixedWidth = 0;
+        for (const column of this.table.columns) {
+            totalColumnFixedWidth += (column.columnInternals.currentPixelWidth
+                !== undefined)
+                ? column.columnInternals.currentPixelWidth
+                : 0;
```

Comment:
I think this can be simplified
```suggestion
            totalColumnFixedWidth += column.columnInternals.currentPixelWidth ?? 0;
```